src/inmanta/agent/forking_executor.py

```diff
@@ -224,6 +246,22 @@ async def call(self, context: ExecutorContext) -> typing.Sequence[inmanta.loader
                 logger.info("Failed to load sources: %s", module_source, exc_info=True)
                 failed.append(module_source)
 
+        if context.server.timer_venv_checkup is not None:
+            context.server.stop_timer_venv_checkup()
+
+        # Only used for testing
+        if self._venv_checkup_interval is not None:
+            context.server.timer_venv_checkup = tornado.ioloop.PeriodicCallback(
+                callback=context.server.touch_inmanta_env_status,
+                callback_time=datetime.timedelta(seconds=self._venv_checkup_interval),
+            )
+        else:
+            context.server.timer_venv_checkup = tornado.ioloop.PeriodicCallback(
+                callback=context.server.touch_inmanta_env_status,
+                callback_time=datetime.timedelta(minutes=1),
+            )
```

Comment:
```suggestion
        interval = self._venv_checkup_interval or 60
        
        context.server.timer_venv_checkup = tornado.ioloop.PeriodicCallback(
            callback=context.server.touch_inmanta_env_status,
            callback_time=datetime.timedelta(seconds=interval),
        )
      
```

Reply:
I would also start the timer before initializing the venv, take decrease time for a race

Reply:
> I would also start the timer before initializing the venv, take decrease time for a race

I'm not sure to see why this would decrease time for a race

Reply:
1. When the executor is starting, up until the point it touches the file, it is racing with the cleanup.
    - I might be wrong, perhaps the entire startup is under lock, keeping the cleanup out? If so, it might be good to document this
    -  Doing it under lock is probably for the best
2. Loading the venv may take a bit of time
3. so writing to the file earlier is better as it reduces the race time